portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns the login URL and status for the user. */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    String redirectUrl = "/";
+    String loginUrl;
+    String userEmail;
+
+    if (userService.isUserLoggedIn()) {
+      userEmail = userService.getCurrentUser().getEmail();
+      loginUrl = userService.createLogoutURL(redirectUrl);
+    } else {
+      userEmail = null;
+      loginUrl = userService.createLoginURL(redirectUrl);
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("text/html;");
+    response.getWriter().println(gson.toJson(new String[] {loginUrl, userEmail}));
```

Comment:
I would opt to define a new object for this rather than put these values in an array. It makes it clearer to the caller and less error prone (is user email index 0 or 1).

Reply:
Is it ok if I just make this change in my next PR? I need to do it anyways so I can add a boolean for whether the user is an admin.

Reply:
Discussed this with Tony in our 1:1